test/shopify_app/controller_concerns/token_exchange_test.rb

```diff
@@ -54,58 +63,28 @@ class TokenExchangeControllerTest < ActionController::TestCase
     @offline_session_id = "offline_#{@shop}"
     @online_session_id = "online_#{@user.id}"
 
-    ShopifyApp.configuration.check_session_expiry_date = false
+    ShopifyApp.configuration.check_session_expiry_date = true
     ShopifyApp.configuration.custom_post_authenticate_tasks = MockPostAuthenticateTasks
   end
 
-  test "Exchanges offline token when session doesn't exist" do
+  test "Exchanges token when session doesn't exist" do
     with_application_test_routes do
       ShopifyAPI::Utils::SessionUtils.stubs(:current_session_id).with(
         @session_token_in_header,
         nil,
         false,
       ).returns(nil, @offline_session_id)
 
-      ShopifyAPI::Auth::TokenExchange.expects(:exchange_token).with(
-        shop: @shop,
-        session_token: @session_token,
-        requested_token_type: ShopifyAPI::Auth::TokenExchange::RequestedTokenType::OFFLINE_ACCESS_TOKEN,
-      ).returns(@offline_session)
+      ShopifyApp::Auth::TokenExchange.expects(:perform).with(@session_token) do
+        ShopifyApp::SessionRepository.store_session(@offline_session)
+      end
 
       ShopifyAPI::Context.expects(:activate_session).with(@offline_session)
 
       get :index, params: { shop: @shop }
     end
   end
 
-  test "Exchanges online token if user session store is configured" do
```

Comment:
Should we keep this test? This logic is still in the token exchange class, right?